code/game/objects/structures/inflatable.dm

```diff
@@ -0,0 +1,238 @@
+	if(take_damage(rand(M.melee_damage_lower, M.melee_damage_upper)))
+		M.visible_message("<span class='danger'>[M] tears open \the [src]!</span>")
+	else
+		M.visible_message("<span class='danger'>[M] [M.attacktext] \the [src]!</span>")
+	M.delayNextAttack(10)
+	return 1
+
+/obj/structure/inflatable/attack_alien(mob/user)
+	user.visible_message("<span class='danger'>[user] rips \the [src] apart!</span>")
+	deflate(1)
+
+/obj/structure/inflatable/proc/take_damage(var/damage, var/sound_effect = 1)
+	health = max(0, health - damage)
+	if(sound_effect)
+		playsound(loc, 'sound/effects/Glasshit.ogg', 75, 1)
+	if(health <= 0)
+		spawn(1)
+			deflate(1)
+		return 1
+	return 0
+
+/obj/structure/inflatable/CtrlClick()
+	hand_deflate()
+
+/obj/structure/inflatable/proc/deflate(var/violent=0, var/deflatespeed = 50)
+	playsound(loc, 'sound/machines/hiss.ogg', 75, 1)
+	var/obj/item/inflatable/remains
+	if(violent)
+		visible_message("[src] rapidly deflates!")
+		remains = new /obj/item/inflatable/torn(loc)
+	else
+		if(!undeploy_path || deflating)
+			return
+		visible_message("\The [src] starts to deflate.")
+		deflating = 1
+		sleep(deflatespeed)
+		visible_message("\The [src] fully deflates.")
+		remains = new undeploy_path(loc)
+	transfer_fingerprints_to(remains)
+	qdel(src)
+
+/obj/structure/inflatable/verb/hand_deflate()
+	set name = "Deflate"
+	set category = "Object"
+	set src in oview(1)
+
+	if(!isliving(usr) || usr.incapacitated() || !usr.Adjacent(src) || !usr.IsAdvancedToolUser() || deflating)
+		return
+
+	verbs -= /obj/structure/inflatable/verb/hand_deflate
+	deflate()
+
+/obj/structure/inflatable/proc/update_nearby_tiles(var/turf/T)
+	if(isnull(air_master))
+		return 0
+
+	if(!T)
+		T = get_turf(src)
+	if(isturf(T))
+		air_master.mark_for_update(T)
+	return 1
+
+/obj/structure/inflatable/Cross(atom/movable/mover, turf/target, height=1.5, air_group = 0)
+	if(air_group)
+		return 0
+	if(istype(mover) && mover.checkpass(PASSGLASS))
+		return 1
+	return !density
+
+/obj/structure/inflatable/door
+	name = "inflatable door"
+
+	icon_state = "door_closed"
+	undeploy_path = /obj/item/inflatable/door
+
+	var/is_open = 0
+	var/busy = 0
+
+/obj/structure/inflatable/door/attack_robot(mob/user)
+	if(user.Adjacent(src))
+		toggle(user)
+
+/obj/structure/inflatable/door/attack_hand(mob/user)
+	toggle(user)
+
+/obj/structure/inflatable/door/proc/toggle(mob/user)
+	if(busy)
+		return
+
+	add_fingerprint(user)
+	if(is_open)
+		Close()
+	else
+		Open()
+	update_nearby_tiles()
+
+/obj/structure/inflatable/door/proc/Open()
+	busy = 1
+	flick("door_opening",src)
+	sleep(5)
+	density = 0
+	is_open = 1
+	update_icon()
+	busy = 0
+
+/obj/structure/inflatable/door/proc/Close()
+	busy = 1
+	flick("door_closing",src)
+	sleep(5)
+	density = 1
+	is_open = 0
+	update_icon()
+	busy = 0
+
+/obj/structure/inflatable/door/update_icon()
+	if(is_open)
+		icon_state = "door_open"
+	else
+		icon_state = "door_closed"
```

Comment:
You could probably use a pretty sick ternary operator to handle this in one line.